BUILD

```diff
@@ -0,0 +1,522 @@
+# ==============  kj ===================
+cc_library(
+    name = "kj",
+    srcs = [
+        "c++/src/kj/arena.c++",
+        "c++/src/kj/array.c++",
+        "c++/src/kj/common.c++",
+        "c++/src/kj/debug.c++",
+        "c++/src/kj/encoding.c++",
+        "c++/src/kj/exception.c++",
+        "c++/src/kj/filesystem.c++",
+        "c++/src/kj/filesystem-disk-unix.c++",
+        "c++/src/kj/filesystem-disk-win32.c++",
+        "c++/src/kj/hash.c++",
+        "c++/src/kj/io.c++",
+        "c++/src/kj/list.c++",
+        "c++/src/kj/main.c++",
+        "c++/src/kj/memory.c++",
+        "c++/src/kj/mutex.c++",
+        "c++/src/kj/parse/char.c++",
+        "c++/src/kj/refcount.c++",
+        "c++/src/kj/source-location.c++",
+        "c++/src/kj/string.c++",
+        "c++/src/kj/string-tree.c++",
+        "c++/src/kj/table.c++",
+        "c++/src/kj/test-helpers.c++",
+        "c++/src/kj/thread.c++",
+        "c++/src/kj/time.c++",
+        "c++/src/kj/units.c++",
+    ] + [
+        "c++/src/kj/miniposix.h",
+        "c++/src/kj/test.h",
+    ],
+    hdrs = [
+        "c++/src/kj/arena.h",
+        "c++/src/kj/array.h",
+        "c++/src/kj/common.h",
+        "c++/src/kj/debug.h",
+        "c++/src/kj/encoding.h",
+        "c++/src/kj/exception.h",
+        "c++/src/kj/filesystem.h",
+        "c++/src/kj/function.h",
+        "c++/src/kj/hash.h",
+        "c++/src/kj/io.h",
+        "c++/src/kj/list.h",
+        "c++/src/kj/main.h",
+        "c++/src/kj/map.h",
+        "c++/src/kj/memory.h",
+        "c++/src/kj/miniposix.h",
+        "c++/src/kj/mutex.h",
+        "c++/src/kj/one-of.h",
+        "c++/src/kj/parse/char.h",
+        "c++/src/kj/parse/common.h",
+        "c++/src/kj/refcount.h",
+        "c++/src/kj/source-location.h",
+        "c++/src/kj/std/iostream.h",
+        "c++/src/kj/string.h",
+        "c++/src/kj/string-tree.h",
+        "c++/src/kj/table.h",
+        "c++/src/kj/thread.h",
+        "c++/src/kj/threadlocal.h",
+        "c++/src/kj/time.h",
+        "c++/src/kj/tuple.h",
+        "c++/src/kj/units.h",
+        "c++/src/kj/vector.h",
+        "c++/src/kj/windows-sanity.h",
+    ],
+    copts = [
+        "-Wno-sign-compare",
+        "-Wno-strict-aliasing",
+        "-Wno-shadow",
+        "-Wno-unused-parameter",
+    ],
+    defines = [
+        "KJ_USE_FUTEX=0",
+    ],
+    includes = ["c++/src"],
```

Comment:
Generally the use of the `includes` attribute is discouraged.

We're builting capnp with Bazel internally and found that builds would fail in some cases where capnp was already installed on the host system.

Lots of

```
In file included from external/com_github_capnproto_capnproto/capnp/compiler/parser.c++:26:
In file included from external/com_github_capnproto_capnproto/capnp/compiler/parser.h:28:
In file included from external/com_github_capnproto_capnproto/capnp/compiler/error-reporter.h:24:
external/com_github_capnproto_capnproto/capnp/compiler/../common.h:148:20: error: redefinition of 'Kind_<unsigned char>'
template <> struct Kind_<uint8_t> { static constexpr Kind kind = Kind::PRIMITIVE; };
                   ^~~~~~~~~~~~~~
/usr/local/include/capnp/common.h:148:20: note: previous definition is here
template <> struct Kind_<uint8_t> { static constexpr Kind kind = Kind::PRIMITIVE; };
                   ^
```

We had to use explicit `isystem` include directives.

```
copts = [
    "-Iexternal/com_github_capnproto_capnproto",
    # ...
]
```

This shouldn't be necessary, but is because capnp is using angled brackets rather than quotes...

```
In file included from external/com_github_capnproto_capnproto/capnp/schema-loader.c++:23:
external/com_github_capnproto_capnproto/capnp/schema-loader.h:25:10: error: 'kj/memory.h' file not found with <angled> include; use "quotes" instead
#include <kj/memory.h>
         ^~~~~~~~~~~~~
         "kj/memory.h"
```

I'll create an issue for this.